test/components/BannerConductor/StateMachine/states/PendingState.spec.ts

```diff
@@ -19,26 +21,36 @@ describe( 'PendingState', function () {
 	} );
 
 	it( 'starts delay timeout on enter', async () => {
-		const setTimeoutSpy = vitest.spyOn( window, 'setTimeout' );
-		const pendingState = new PendingState( page, 60, 1 );
+		let called = false;
+		let delay = 0;
+
+		const timerSpy = newTimerSpy();
+		timerSpy.setTimeout = ( fn: () => void, after: number ): number => {
+			called = true;
+			delay = after;
+			fn();
+			return 0;
+		};
```

Comment:
I'd prefer the `TimerSpy` to collect that information in the implementation (even if it's only needed in this test).